tests/rptest/tests/archival_test.py

```diff
@@ -0,0 +1,661 @@
+            cloud_storage_api_endpoint_port=9000,
+            cloud_storage_reconciliation_interval_ms=1000,
+            cloud_storage_max_connections=5,
+            log_segment_size=1048576  # 1MB
+        )
+        super(ArchivalTest, self).__init__(test_context=test_context,
+                                           extra_rp_conf=extra_rp_conf)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.s3_client = S3Client(
+            region='panda-region',
+            access_key=u"panda-user",
+            secret_key=u"panda-secret",
+            endpoint=f'http://{ArchivalTest.s3_host_name}:9000',
+            logger=self.logger)
+
+    def setUp(self):
+        self.s3_client.empty_bucket(ArchivalTest.s3_bucket_name)
+        self.s3_client.create_bucket(ArchivalTest.s3_bucket_name)
+        # Deletes in S3 are eventually consistent so we might still
+        # see previously removed objects for a while.
+        validate(self._check_bucket_is_emtpy, self.logger, 300)
+        super().setUp()
+
+    def tearDown(self):
+        self.s3_client.empty_bucket(ArchivalTest.s3_bucket_name)
+        super().tearDown()
+
+    @cluster(num_nodes=3)
+    def test_write(self):
+        """Simpe smoke test, write data to redpanda and check if the
+        data hit the S3 storage bucket"""
+        self.kafka_tools.produce(self.topic, 10000, 1024)
+        validate(self._quick_verify, self.logger, 60)
+
+    @cluster(num_nodes=3)
+    def test_isolate(self):
+        """Verify that our isolate/rejoin facilities actually work"""
+        with firewall_blocked(self.redpanda.nodes, self._get_s3_endpoint_ip()):
+            self.kafka_tools.produce(self.topic, 10000, 1024)
+            time.sleep(10)  # can't busy wait here
+            self._check_bucket_is_emtpy()
+
+    @cluster(num_nodes=3)
+    def test_reconnect(self):
+        """Disconnect redpanda from S3, write data, connect redpanda to S3
+        and check that the data is uploaded"""
+        with firewall_blocked(self.redpanda.nodes, self._get_s3_endpoint_ip()):
+            self.kafka_tools.produce(self.topic, 10000, 1024)
+            time.sleep(5)  # sleep is needed because we need to make sure that
+            # reconciliation loop kicked in and started uploading
+            # data, otherwse we can rejoin before archival storage
+            # will even try to upload new segments
+        validate(self._quick_verify, self.logger, 60)
+
+    @cluster(num_nodes=3)
+    def test_one_node_reconnect(self):
+        """Disconnect one redpanda node from S3, write data, connect redpanda to S3
+        and check that the data is uploaded"""
+        self.kafka_tools.produce(self.topic, 1000, 1024)
+        leaders = list(self._get_partition_leaders().values())
+        with firewall_blocked(leaders[0:1], self._get_s3_endpoint_ip()):
+            self.kafka_tools.produce(self.topic, 9000, 1024)
+            time.sleep(5)  # sleep is needed because we need to make sure that
```

Comment:
if this sleep is not long enough (and we rejoin before uploads start) will the test pass or fail?

another trick I've used is that the `self.redpanda` service lets you query for Prometheus metrics. So you can add some checks for the events you care about to avoid sleep.